go/vt/hook/hook_test.go

```diff
@@ -38,3 +45,223 @@ func TestExecuteContext(t *testing.T) {
 	hr = h.Execute()
 	assert.Equal(t, HOOK_SUCCESS, hr.ExitStatus)
 }
+
+func TestExecuteOptional(t *testing.T) {
+	vtroot, err := vtenv.VtRoot()
+	require.NoError(t, err)
+
+	echo, err := exec.LookPath("echo")
+	require.NoError(t, err)
+
+	echoHookPath := path.Join(vtroot, "vthook", "echo")
+
+	if _, err := os.Lstat(echoHookPath); err == nil {
+		require.NoError(t, os.Remove(echoHookPath))
+	}
+
+	require.NoError(t, os.Symlink(echo, echoHookPath))
+	defer func() {
+		require.NoError(t, os.Remove(echoHookPath))
+	}()
+	tt := []struct {
+		name          string
+		hookName      string
+		parameters    []string
+		expectedError string
+	}{
+		{
+			name:       "HookSuccess",
+			hookName:   "echo",
+			parameters: []string{"test"},
+		},
+		{
+			name:          "HookDoesNotExist",
+			hookName:      "nonexistent-hook",
+			parameters:    []string{"test"},
+			expectedError: "",
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			h := NewHook(tc.hookName, tc.parameters)
+			err := h.ExecuteOptional()
+			if tc.expectedError == "" {
+				assert.NoError(t, err)
+			} else {
+				assert.Error(t, err)
+				assert.Contains(t, err.Error(), tc.expectedError)
```

Comment:
you can directly use `assert.ErrorContains` instead of using two distinct functions here